src/MSBuild/XMake.cs

```diff
@@ -4406,10 +4414,35 @@ private static LoggerDescription ParseLoggingParameter(string parameter, string
             string binaryLogFilePath,
             ILogger[] loggers,
             IEnumerable<DistributedLoggerRecord> distributedLoggerRecords,
-            int cpuCount)
+            int cpuCount,
+            bool isBuildCheckEnabled,
+            ISet<string> warningsAsErrors,
+            ISet<string> warningsNotAsErrors,
+            ISet<string> warningsAsMessages)
         {
             var replayEventSource = new BinaryLogReplayEventSource();
 
+            ILogger buildCheckLogger = null;
+
+            if (isBuildCheckEnabled)
+            {
+                List<ForwardingLoggerRecord> remoteLoggerRecords = new List<ForwardingLoggerRecord>();
+                foreach (DistributedLoggerRecord distRecord in distributedLoggerRecords)
+                {
+                    remoteLoggerRecords.Add(new ForwardingLoggerRecord(distRecord.CentralLogger, distRecord.ForwardingLoggerDescription));
+                }
+
+                BuildManager.DefaultBuildManager.AttachBuildCheckForReplay(
+                    loggers,
+                    remoteLoggerRecords,
+                    warningsAsErrors,
+                    warningsNotAsErrors,
+                    warningsAsMessages,
+                    out buildCheckLogger);
+
+                buildCheckLogger.Initialize(replayEventSource);
```

Comment:
Let's discuss offline - but I believe we'll need to decorate rather than attach. IoW - `replayEventSource` will be consumed by the BuildCheck that will produce a different ReplayEventSource - that way it'll be able to inject its own events